src/AWSS3.jl

```diff
@@ -1145,6 +1178,70 @@ function s3_multipart_upload(
     return s3_complete_multipart_upload(aws, upload, tags; parse_response, kwargs...)
 end
 
+"""
+    s3_multipart_copy(aws::AbstractAWSConfig, bucket, path; to_bucket=bucket, to_path=path,
+                      part_size_mb=50, version=nothing, parse_response::Bool=true,
+                      kwargs...)
+
+Copy the object at `path` in `bucket` to `to_path` in `to_bucket` using a
+[multipart copy](https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html).
+
+# Optional Arguments
+- `part_size_mb`: maximum size per uploaded part, in mebibytes (MiB).
```

Comment:
I wonder if it's worth exposing an option that allows matching the part size between the source and destination. IIUC, that should make the range-based accesses faster while copying. If a file is big enough for a multipart copy, it was probably uploaded with a multipart upload, in which case the parts and their sizes can be obtained with `S3.get_object_attributes`. Lacking that permission, one can also get the part size with `S3.head_object` by passing `Dict("partNumber" => 1)` as a query parameter, and the number of parts will be in the entity tag of the source object.